### PR TITLE
fixing the propagation of allow_{upsample,downsample}

### LINKED ISSUE #2053: FrameExtractionOptions::allow_downsample ignored in online2-wav-nnet3-latgen-faster
When you try to run a model trained on 8kHz data on a 16kHz waveform in `online2-wav-nnet3-latgen-faster` even with `--allow-downsample=true` in your mfcc.conf (or plp.conf or filter_bank.conf, or whatever configuration file the online.conf output by `steps/online/nnet3/prepare_online_decoding.sh` points to), the program will crash.

Log looks like this:

```
online2-wav-nnet3-latgen-faster --online=false --do-endpointing=false --frame-subsampling-factor=3 --config=exp/decode_with_model/1a/online_model/conf/online.conf --max-active=7000 --beam=15.0 --lattice-beam=8.0 --acoustic-scale=1.0 --word-symbol-table=exp/decode_with_model/1a/graph/words.txt --ivector-silence-weighting.silence-phones=1:2:3:4:5:6:7:8:9:10:11:12:13:14:15:16:17:18:19:20 /home/galv/development/kaldi-models/aspire/s5/exp/chain/tdnn_7b/final.mdl exp/decode_with_model/1a/graph/HCLG.fst ark:data/all/split27utt/9/spk2utt 'ark:extract-segments scp:data/all/split27utt/9/wav.scp data/all/split27utt/9/segments ark:- |' ark:exp/decode_with_model/1a/decode/lattices/9.ark
LOG (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:SelectGpuId():cu-device.cc:110) Manually selected to compute on CPU.
LOG (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:ComputeDerivedVars():ivector-extractor.cc:183) Computing derived variables for iVector extractor
LOG (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:ComputeDerivedVars():ivector-extractor.cc:204) Done.
LOG (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:RemoveSomeNodes():nnet-nnet.cc:922) Removed 2 orphan nodes.
LOG (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:RemoveOrphanComponents():nnet-nnet.cc:847) Removing 2 orphan components.
LOG (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:Collapse():nnet-utils.cc:797) Added 1 components, removed 2
LOG (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:CompileLooped():nnet-compile-looped.cc:336) Spent 0.082777 seconds in looped compilation.
extract-segments scp:data/all/split27utt/9/wav.scp data/all/split27utt/9/segments ark:-
ERROR (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:AcceptWaveform():online-feature.cc:45) Sampling frequency mismatch, expected 8000, got 16000

[ Stack-Trace: ]
online2-wav-nnet3-latgen-faster() [0x109faba]
kaldi::MessageLogger::HandleMessage(kaldi::LogMessageEnvelope const&, char const*)
kaldi::MessageLogger::~MessageLogger()
kaldi::OnlineGenericBaseFeature<kaldi::MfccComputer>::AcceptWaveform(float, kaldi::VectorBase<float> const&)
kaldi::OnlineNnet2FeaturePipeline::AcceptWaveform(float, kaldi::VectorBase<float> const&)
main
__libc_start_main
_start

WARNING (online2-wav-nnet3-latgen-faster[5.2.204~36-8d035]:Close():kaldi-io.cc:501) Pipe extract-segments scp:data/all/split27utt/9/wav.scp data/all/split27utt/9/segments ark:- | had nonzero return status 13
# Accounting: time=5 threads=1
# Ended (code 255) at Wed Nov 29 09:14:29 PST 2017, elapsed time 5 seconds
```

This is because the main() function of `online2-wav-nnet3-latgen-faster` assumes that the sampling frequency of the input waveform is the desired sampling frequency (see the line `BaseFloat samp_freq = wave_data.SampFreq();`), without first checking allow_downsample in FrameExtractionOptions (which is transitively held by the OnlineNnet2FeaturePipeline via OnlineGenericBaseFeature). Unfortunately, this value is not exposed through any public interface that I know of, so it's not easy to fix this.

Work around is to downsample the waveform in advance, e.g., with sox.

### PR DESCRIPTION
**Pull request by jtrmal**
Should resolve #2053
Gave it a short test on yesno and didn't see any problems

### CODE REVIEW

File: src/feat/resample.h [hunk 1]
```diff
@@ -185,6 +185,10 @@ class LinearResample {
   /// Resample(x, y, true) for the last piece.  Call it unnecessarily between
   /// signals will not do any harm.
   void Reset();
+
+  //// Return the input and output sampling rates (for checks, for example)
+  inline int32 GetInputSamplingRate() {return samp_rate_in_;};
```
Comment (danpovey) [inline src/feat/resample.h]: can you add spaces in side the {}?  I think the final semicolon is not required.

**Comment by jtrmal** [comment]
done.
y.

**Comment by danpovey** [comment]
thanks, will merge when checks done

**Comment by bayethiernodiop** [comment]
Thx a lot for this fix really appreciated

**Pull request merged**